server.go

```diff
@@ -22,6 +22,34 @@ const (
 	SftpServerWorkerCount = 8
 )
 
+// AllocationMode defines allocation modes
+type AllocationMode uint8
```

Comment:
When defining enums use the type `int` (which should have the most efficient size for the architecture).

Bytes are not any faster to use than `int`s, and rarely save any space as well, due to memory alignment.